subprojects/ide/src/main/java/org/gradle/plugins/ide/eclipse/model/Project.java

```diff
@@ -242,6 +259,63 @@ private void addLinkedResourcesToXml() {
         }
     }
 
+    private void addResourceFiltersToXml() {
+        // other nodes like linkedResources are appended even if they are empty, but as
+        // resourceFilters are newly-added, we do not append an empty tag by default to
+        // avoid changing existing .project files unnecessarily
+        if (resourceFilters == null || resourceFilters.size() == 0) {
+            return;
+        }
+        Node parent = getXml().appendNode("filteredResources");
+        for (ResourceFilter resourceFilter : resourceFilters) {
+            Node filterNode = parent.appendNode("filter");
+            filterNode.appendNode("id", resourceFilter.getId());
+            int type = getResourceFilterType(resourceFilter);
+            filterNode.appendNode("type", type);
+            filterNode.appendNode("name"); // always empty
+            Node matcherNode = filterNode.appendNode("matcher");
+            ResourceFilterMatcher matcher = resourceFilter.getMatcher();
+            matcherNode.appendNode("id", matcher.getId());
+            if (!isNullOrEmpty(matcher.getArguments())) {
+                matcherNode.appendNode("arguments", matcher.getArguments());
+            }
+        }
+    }
+
+    private int getResourceFilterType(ResourceFilter resourceFilter) {
+        int type = 0;
+        if (resourceFilter == null) {
```

Comment:
Ok, I will pick and document some default values and prevent nulls in the ResourceFilter setters.